### PR TITLE
Switch to https

### LINKED ISSUE #3337: Reference redirects to different link
This link redirects to a https variant with `301 Moved Permanently`, so we should update it:

https://github.com/modelica/ModelicaSpecification/blob/6f3779266dc5c35edf9e2a03c19add0770d035de/mls.bib#L10

```
$ curl --verbose http://epubs.siam.org/doi/abs/10.1137/0909014
*   Trying 104.18.31.167:80...
* Connected to epubs.siam.org (104.18.31.167) port 80 (#0)
> GET /doi/abs/10.1137/0909014 HTTP/1.1
> Host: epubs.siam.org
> User-Agent: curl/7.85.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 301 Moved Permanently
< Date: Mon, 06 Feb 2023 08:26:42 GMT
< Transfer-Encoding: chunked
< Connection: keep-alive
< Cache-Control: max-age=3600
< Expires: Mon, 06 Feb 2023 09:26:42 GMT
< Location: https://epubs.siam.org/doi/abs/10.1137/0909014
```

### PR DESCRIPTION
**Pull request by HansOlsson**
Closes #3337

### CODE REVIEW

**Comment by HansOlsson** [comment]
Just updating the link in the bibliography

**Review by maltelenz — APPROVED** [review]

**Comment by sjoelund** [comment]
Could have also been changed to https://doi.org/10.1137/0909014 (the permanent link)

**Comment by HansOlsson** [comment]
> Could have also been changed to https://doi.org/10.1137/0909014 (the permanent link)

That's also possible, but I tried to read up and even if working there are some claims that it should be displayed differently (as [doi:10.1137/0909014](https://doi.org/10.1137/0909014) instead of as [https://doi.org/10.1137/0909014](https://doi.org/10.1137/0909014) ), while others prefer the simpler link - so I then postponed that.

**Comment by sjoelund** [comment]
Their own webpage lists it as https:// but either is fine as long as we are consistent

**Comment by sjoelund** [comment]
It's also possible to in bibtex use `doi = {10......}` and it typesets it consistently

**Comment by HansOlsson** [comment]
> Their own webpage lists it as https:// but either is fine as long as we are consistent

That made me try to see if there are other links that should be doi - and it seems messy, and no other good examples were found.

ColacoPouzet2003ClocksFirstClass is the same as doi:10.1007/978-3-540-45212-6_10 - except that you need a subscription for the latter. However, I guess that at some point we will get link rot so we should use the doi as well.

https://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.151.1745 has a link to "Nonlinear Inverse Models for Control" and https://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.127.3787 has a link to "Inline Integration: A new mixed symbolic/numeric approach for solving differential-algebraic equation systems" - but the given doi aren't working.

File: mls.bib [hunk 1]
```diff
@@ -7,7 +7,7 @@ @Article{Pantelides1988ConsistentInitialization
   volume = {9},
   number = {2},
   pages = {213–-231},
-  url = {http://epubs.siam.org/doi/abs/10.1137/0909014},
+  url = {https://epubs.siam.org/doi/abs/10.1137/0909014},
```
Comment (HansOlsson) [inline mls.bib:10]:
```suggestion
  url = {https://doi.org/10.1137/0909014},
```
Ok switch to doi

**Comment by sjoelund** [comment]
> https://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.151.1745 has a link to "Nonlinear Inverse Models for Control" and https://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.127.3787 has a link to "Inline Integration: A new mixed symbolic/numeric approach for solving differential-algebraic equation systems" - but the given doi aren't working.

Yeah, citeseer isn't a publisher and is not really something you want to link to if you have a choice...

**Pull request merged**